components/sensor/sht3xd/sht3xd.go

```diff
@@ -0,0 +1,182 @@
+// Package sht3xd implements a sht3x-d sensor for temperature and humidity
+package sht3xd
+
+import (
+	"context"
+	"encoding/binary"
+	"fmt"
+
+	"github.com/edaniels/golog"
+	"go.uber.org/multierr"
+	"go.viam.com/utils"
+
+	"go.viam.com/rdk/components/board"
+	"go.viam.com/rdk/components/generic"
+	"go.viam.com/rdk/components/sensor"
+	"go.viam.com/rdk/config"
+	"go.viam.com/rdk/registry"
+	"go.viam.com/rdk/resource"
+	rdkutils "go.viam.com/rdk/utils"
+)
+
+var modelname = resource.NewDefaultModel("sht3xd")
+
+const (
+	defaultI2Caddr = 0x44
+	// Addresses of sht3xd registers.
+	sht3xdCOMMANDSOFTRESET1 = 0x30
+	sht3xdCOMMANDSOFTRESET2 = 0xA2
+	sht3xdCOMMANDPOLLINGH1  = 0x24
+	sht3xdCOMMANDPOLLINGH2  = 0x00
+)
+
+// AttrConfig is used for converting config attributes.
+type AttrConfig struct {
+	Board   string `json:"board"`
+	I2CBus  string `json:"i2c_bus"`
+	I2cAddr int    `json:"i2c_addr,omitempty"`
+}
+
+// Validate ensures all parts of the config are valid.
+func (config *AttrConfig) Validate(path string) ([]string, error) {
+	var deps []string
+	if len(config.Board) == 0 {
+		return nil, utils.NewConfigValidationFieldRequiredError(path, "board")
+	}
+	deps = append(deps, config.Board)
+	if len(config.I2CBus) == 0 {
+		return nil, utils.NewConfigValidationFieldRequiredError(path, "i2c bus")
```

Comment:
[nit] we like having the field match the attribute field exactly, so `i2c_bus`, in this case.